presto-atop/src/main/java/io/prestosql/plugin/atop/AtopConnector.java

```diff
@@ -50,6 +50,13 @@ public AtopConnector(
         this.accessControl = requireNonNull(accessControl, "accessControl is null");
     }
 
+    @Override
+    public boolean isSingleStatementWritesOnly()
+    {
+        // Actually, writes are not supported at all.
```

Comment:
I dislike all of these comments. They seem puerile, and seem to indicate that the API is wrong. Instead, we should change the default implementation to
```java
throw new PrestoException(NOT_SUPPORTED, "This connector does not support writes");
```
This solves the problem of not having the "wrong" behavior by default, removes useless code from connectors, and is aligned with other `ConnectorMetadata` methods like `beginDelete()`.

Reply:
I agree the API was wrong, hence the issue (https://github.com/prestosql/presto/issues/5468).
The current state is not my ideal end state. I want to make the method have a _different_ default implementation.

Reply:
@electrum thoughts on that?

Reply:
@electrum any thoughts about this?